src/rules/variableNameRule.ts

```diff
@@ -60,130 +62,136 @@ export class Rule extends Lint.Rules.AbstractRule {
         type: "style",
         typescriptOnly: false,
     };
-    /* tslint:enable:object-literal-sort-keys */
 
     public static FORMAT_FAILURE = "variable name must be in camelcase or uppercase";
     public static KEYWORD_FAILURE = "variable name clashes with keyword/type";
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        const variableNameWalker = new VariableNameWalker(sourceFile, this.getOptions());
-        return this.applyWithWalker(variableNameWalker);
+        return this.applyWithFunction(sourceFile, (ctx) => walk(ctx, parseOptions(this.getOptions().ruleArguments)));
```

Comment:
and as always, why not pass options as third parameter to `applyWithFunction` and access it via `ctx.options`?
You would save a closure and a (unnecessary) function call.